Classes/Finishers/RedirectFinisher.php

```diff
@@ -55,18 +65,18 @@ public function executeInternal()
 
         $uriParts = parse_url($uri);
         if (!isset($uriParts['scheme']) || $uriParts['scheme'] === '') {
-            $uri = $request->getHttpRequest()->getBaseUri() . $uri;
+            $uri = $request->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI) . $uri;
         }
 
         $escapedUri = htmlentities($uri, ENT_QUOTES, 'utf-8');
 
         $response = $formRuntime->getResponse();
 
         $response->setContent('<html><head><meta http-equiv="refresh" content="' . $delay . ';url=' . $escapedUri . '"/></head></html>');
-        $response->setStatus($statusCode);
+        $response->setStatusCode($statusCode);
 
         if ($delay === 0) {
-            $response->setHeader('Location', (string)$uri);
+            $response->setRedirectUri($this->uriFactory->createUri((string)$uri));
```

Comment:
Do we really need a factory for this?
Internally it's just `return new Uri($uri)` anyways

Reply:
As far as I understood from @kitsunet, that makes the URI implementation exchangeable. I don't know if its best practice or not, there are not much usages yet ;)

Reply:
Makes sense because we can swap the factory implementation. Cheers! :)